narwhals/utils.py

```diff
@@ -2000,3 +2003,35 @@ def to_tuple(self) -> tuple[_T, ...]:
         # Collect and return as a `tuple`.
         it = self._into_iter()
         return it if isinstance(it, tuple) else tuple(it)
+
+
+def parse_interval_string(every: str) -> tuple[str, str]:
+    """Parse a string like "1d", "2h", "3m" into a tuple of (number, unit).
+
+    Returns:
+        A tuple of number and unit parsed from the interval string.
+    """
+    if match := _PATTERN_INTERVAL.match(every):
+        return match["number"], match["unit"]
+    msg = f"Invalid frequency string: {every}."
+    raise ValueError(msg)
+
+
+def unit_to_str() -> dict[str, str]:
+    """Convert time unit to string.
+
+    Returns:
+        A dictionary mapping time unit to strings used by DuckDB and PyArrow.
+    """
+    return {
+        "y": "year",
+        "q": "quarter",
+        "mo": "month",
+        "d": "day",
+        "h": "hour",
+        "m": "minute",
+        "s": "second",
+        "ms": "millisecond",
+        "us": "microsecond",
+        "ns": "nanosecond",
+    }
```

Comment:
could this dictionary just be a constant in `_duckdb/utils.py` and `_arrow/utils.py`?